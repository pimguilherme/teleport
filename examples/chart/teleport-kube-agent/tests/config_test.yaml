suite: ConfigMap
templates:
  - config.yaml
tests:
  - it: matches snapshot for affinity.yaml
    values:
      - ../.lint/affinity.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for all-v5.yaml
    values:
      - ../.lint/all-v5.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for all-v6.yaml
    values:
      - ../.lint/all-v6.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot and tests for annotations.yaml
    values:
      - ../.lint/annotations.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - equal:
          path: metadata.annotations.kubernetes\.io/config
          value: test-annotation
      - equal:
          path: metadata.annotations.kubernetes\.io/config-different
          value: 2
      - matchSnapshot: {}

  - it: matches snapshot for aws-databases.yaml
    values:
      - ../.lint/aws-databases.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for backwards-compatibility.yaml
    values:
      - ../.lint/backwards-compatibility.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: does not generate a config for clusterrole.yaml
    values:
      - ../.lint/clusterrole.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for db.yaml
    values:
      - ../.lint/db.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for imagepullsecrets.yaml
    values:
      - ../.lint/imagepullsecrets.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for initcontainers.yaml
    values:
      - ../.lint/initcontainers.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for log-basic.yaml
    values:
      - ../.lint/log-basic.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for log-extra.yaml
    values:
      - ../.lint/log-extra.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for log-legacy.yaml
    values:
      - ../.lint/log-legacy.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for node-selector.yaml
    values:
      - ../.lint/node-selector.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for pdb.yaml
    values:
      - ../.lint/log-extra.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: does not generate a config for pdb.yaml
    values:
      - ../.lint/pdb.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for resources.yaml
    values:
      - ../.lint/resources.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for stateful.yaml
    values:
      - ../.lint/stateful.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for tolerations.yaml
    values:
      - ../.lint/tolerations.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}

  - it: matches snapshot for volumes.yaml
    values:
      - ../.lint/volumes.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ConfigMap
      - matchSnapshot: {}
