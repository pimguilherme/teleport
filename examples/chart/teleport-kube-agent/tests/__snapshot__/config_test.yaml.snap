does not generate a config for clusterrole.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
does not generate a config for pdb.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster-name

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot and tests for annotations.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      annotations:
        kubernetes.io/config: test-annotation
        kubernetes.io/config-different: 2
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for affinity.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for all-v5.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster-name
          labels:
            cluster: testing

        app_service:
          enabled: true
          apps:
            - labels:
                environment: test
              name: grafana
              uri: http://localhost:3000

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for all-v6.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster-name
          labels:
            cluster: testing

        app_service:
          enabled: true
          apps:
            - labels:
                environment: test
              name: grafana
              uri: http://localhost:3000

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      annotations:
        kubernetes.io/config: test-annotation
        kubernetes.io/config-different: 2
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for aws-databases.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: false

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for backwards-compatibility.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster-name

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for db.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: false

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for imagepullsecrets.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for initcontainers.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for log-basic.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: json
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster-name

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for log-extra.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: DEBUG
            output: /var/lib/teleport/test.log
            format:
              output: json
              extra_fields: ["level","timestamp","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster-name

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for log-legacy.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: DEBUG
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster-name

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for node-selector.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster-name

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for pdb.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: DEBUG
            output: /var/lib/teleport/test.log
            format:
              output: json
              extra_fields: ["level","timestamp","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster-name

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for resources.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for stateful.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster-name

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for tolerations.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
matches snapshot for volumes.yaml:
  1: |
    apiVersion: v1
    data:
      teleport.yaml: |
        teleport:
          auth_token: "/etc/teleport-secrets/auth-token"
          auth_servers: ["proxy.example.com:3080"]
          log:
            severity: INFO
            output: stderr
            format:
              output: text
              extra_fields: ["timestamp","level","component","caller"]

        kubernetes_service:
          enabled: true
          kube_cluster_name: test-kube-cluster

        app_service:
          enabled: false

        auth_service:
          enabled: false
        ssh_service:
          enabled: false
        proxy_service:
          enabled: false
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
